### PR TITLE
Add hover tooltip on sorting disabled columns

### PR DESCRIPTION
**Pull request by Droniu**
Cherry pick of #1784 

For review - gift cards part by @Cloud11PL, my commits have been reviewed already in #1784 

**PR intended to be tested with API branch:** 3.1

### Screenshots
image

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://master.staging.saleor.cloud/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-3637 Add hover tooltip on sorting disabled columns](https://app.clickup.com/t/2549495/SALEOR-3637)

File: src/giftCards/GiftCardsList/GiftCardsListTable/GiftCardsListTableHeader/GiftCardsListTableHeader.tsx [hunk 1]
```diff
@@ -68,19 +71,24 @@ const GiftCardsListTableHeader: React.FC<GiftCardsListTableHeaderProps> = ({
...
+    options: {
+      className: classes.colBalance,
+      textAlign: "right"
+    },
+    onClick: () => onSort(GiftCardUrlSortField.balance),
+    direction: getDirection(GiftCardUrlSortField.balance),
+    disabled: !canBeSorted(GiftCardUrlSortField.balance, isCurrencySelected),
+    dependency: filterLabels.currencyLabel.defaultMessage
+  };
+
```
Comment (Droniu) [inline src/giftCards/GiftCardsList/GiftCardsListTable/GiftCardsListTableHeader/GiftCardsListTableHeader.tsx:91]:
```suggestion
  const headerTooltipItem: HeaderItem & {
    tooltip: string | React.ReactNodeArray;
    disabled: boolean;
  } = {
    title: messages.giftCardsTableColumnBalanceTitle,
    options: {
      className: classes.colBalance,
      textAlign: "right"
    },
    onClick: () => onSort(GiftCardUrlSortField.balance),
    direction: getDirection(GiftCardUrlSortField.balance),
    disabled: !canBeSorted(GiftCardUrlSortField.balance, isCurrencySelected),
    tooltip: intl.formatMessage(commonTooltipMessages.noFilterSelected, {
      filterName: <b>{filterLabels.currencyLabel.defaultMessage}</b>
    })
  };
  const {
    title: headerTooltipItemTitle,
    ...headerTooltipItemProps
  } = headerTooltipItem;
```

File: src/giftCards/GiftCardsList/GiftCardsListTable/GiftCardsListTableHeader/GiftCardsListTableHeader.tsx [hunk 2]
```diff
@@ -105,18 +113,27 @@ const GiftCardsListTableHeader: React.FC<GiftCardsListTableHeaderProps> = ({
...
+          {...omit(headerTooltipItem, ["title"])}
+          tooltip={intl.formatMessage(commonTooltipMessages.noFilterSelected, {
+            filterName: <b>{headerTooltipItem.dependency}</b>
+          })}
+        >
+          <Label
+            text={intl.formatMessage(headerTooltipItem.title)}
+            size={LabelSizes.md}
+          />
+        </TooltipTableCellHeader>
```
Comment (Droniu) [inline src/giftCards/GiftCardsList/GiftCardsListTable/GiftCardsListTableHeader/GiftCardsListTableHeader.tsx:136]:
```suggestion
        <TooltipTableCellHeader {...headerTooltipItemProps}>
          <Label
            text={intl.formatMessage(headerTooltipItemTitle)}
            size={LabelSizes.md}
          />
        </TooltipTableCellHeader>
```

File: src/giftCards/GiftCardsList/GiftCardsListTable/GiftCardsListTableHeader/GiftCardsListTableHeader.tsx [hunk 3]
```diff
@@ -5,13 +5,17 @@ import TableCellHeader, {
   TableCellHeaderProps
 } from "@saleor/components/TableCellHeader";
 import TableHead from "@saleor/components/TableHead";
+import TooltipTableCellHeader from "@saleor/components/TooltipTableCellHeader";
+import { commonTooltipMessages } from "@saleor/components/TooltipTableCellHeader/messages";
 import Label, {
   LabelSizes
 } from "@saleor/orders/components/OrderHistory/Label";
 import { getArrowDirection } from "@saleor/utils/sort";
+import omit from "lodash/omit";
```
Comment (Droniu) [inline src/giftCards/GiftCardsList/GiftCardsListTable/GiftCardsListTableHeader/GiftCardsListTableHeader.tsx:14]:
```suggestion
```

**Review by jwm0 — APPROVED** [review]

**Pull request merged**